lib/parsers/yarn2-lock-parse.ts

```diff
@@ -0,0 +1,56 @@
+import * as _ from 'lodash';
+
+import {LockfileType} from './';
+import getRuntimeVersion from '../get-node-runtime-version';
+import {
+  InvalidUserInputError,
+  UnsupportedRuntimeError,
+} from '../errors';
+import {YarnLockBase, YarnLockDeps} from './yarn-lock-parse-base';
+import {YarnLockParseBase} from './yarn-lock-parse-base';
+import {YarnLockFileKeyNormalizer, yarnLockFileKeyNormalizer} from './yarn-utils';
+
+export type Yarn2Lock  = YarnLockBase<LockfileType.yarn2>;
+
+export class Yarn2LockParser extends YarnLockParseBase<LockfileType.yarn2> {
+
+  private yarnLockfileParser;
+  private keyNormalizer: YarnLockFileKeyNormalizer;
+
+  constructor() {
+    super(LockfileType.yarn2);
+    // @yarnpkg/parsers doesn't work with Node.js < 10
+    if (getRuntimeVersion() < 10) {
+      throw new UnsupportedRuntimeError('yarn.lock parsing is supported for ' +
+        'Node.js v10 and higher.');
+    }
+    this.yarnLockfileParser = require('@yarnpkg/parsers');
+    const structUtils = require('@yarnpkg/core').structUtils;
```

Comment:
Why not `import` (if it's intended, like lazy loading or similar, better add a comment to explain why)?

Reply:
Also note that the `@yarnpkg/parsers` is really just a super light wrapper around `js-yaml`. Might make sense to use it directly, your call.

Reply:
Thought about using straight yaml parser but wasn't sure it is enough. I will replace it. 
About the lazy loading it is mentioned in a few places. This library still needs to support node 8 and above